.github/workflows/python_wheels_macos.yml

```diff
@@ -0,0 +1,53 @@
+name: Build Python wheels targeting MacOS
+
+on: [push, pull_request]
+
+permissions:
+  contents: read  # to fetch code (actions/checkout)
+
+defaults:
+  run:
+    shell: bash -l {0}
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+env:
+  BRANCH_NAME: >-
+    ${{ github.event.pull_request.number && 'PR-' }}${{ github.event.pull_request.number || github.ref_name }}
+
+jobs:
+  python-wheels-macos:
+    name: Build wheel for ${{ matrix.platform_id }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+        - os: macos-13
+          platform_id: macosx_x86_64
+        - os: macos-14
+          platform_id: macosx_arm64
```

Comment:
Commenting here as one example of something I'm confused about.... am I right that the jobs in this `python_wheels_macos.yml` workflow are running on GitHub-hosted runners, not anything managed by RunsOn?

If so, how is this new file related to this PR's goals?

I'm confused enough by this that I'm worried I'm misunderstanding something foundational about this PR 😬

Reply:
We have a mix of jobs, some using self-hosted runners and others using Microsoft-hosted runners. I will update the doc to explain how self-hosted runners are indicated in the workflow YAML syntax.

Reply:
I understand that there will be a mix.

But how is this new file related to this PR's goals? It seems to only contain jobs that run on Microsoft-hosted runners, and looks almost identical to what already exists on `master`:

https://github.com/dmlc/xgboost/blob/bb2e701bf30fe75a0a4b48e599a7d64020325727/.github/workflows/python_wheels.yml#L15-L26

Reply:
Not sure why the diff is larger. Essentially `python_wheels_macos.yml` is a renamed version of `python_wheels.yml`. I rename the workflow to make it clearer about what it does.

Reply:
Ok, I see. I probably would have left such changes out of this PR and #11001, to help reviewers focus on the core RunsOn-related changes.